runtime/gateway.go

```diff
@@ -127,6 +127,7 @@ type Gateway struct {
 	tracerCloser          io.Closer
 
 	requestUUIDHeaderKey string
+	isUnhealthy          bool
```

Comment:
should be atomic or its read/write protected by a mutex

Reply:
good point.